### ISSUE TITLE
moved from heroku to google fine but looking to find the data api

### ISSUE THREAD

**Issue by osheagrainne** [NightscoutFoundation/xDrip#2514]
Hi all
Ive moved over with no issues and thank you for the clear instructions. My boyfriend is also able to follow by using the token etc.

I'm not overly technical hence why I am here. I use a fitbit for my hypo alerts as its the only thing that wakes me. The clock face that i use from pedantic avenger asks for the DATA API URL. His files haven't been updated yet so when I try using the old format from Heroku but with the new host site it wont allow me. I would appreciate any help if it can be done.

Thanks in advance.

**Comment by Navid200** [comment]
I can make a suggestion.

You can change a variable that I suspect will let fitbit behave as it did before.  However, this may result in cost for Google Cloud.  I have no idea how much.  And it may not even increase cost.  We are still gathering data.

I suggest you make the change just to see if it enables the watch or not.  If it  doesn't you can change it back to the way it was because it is pointless.
If it makes the watch work fine, we can wait for more input from others and decide how to handle this.

If you like to try this, you need to change one variable.
The variable name is the following.
AUTH_DEFAULT_ROLES

The value it is set to now on your system must be: denied

If you want to try the experiment, change it to: readable

After you make the change, please don't forget to restart the server (restart Nightscout).  There is an option on the manu to restart.  After you restart, you need to wait 1 minute.  Then, you can try Nightscout again and see if the watch works.

If it still doesn't work, please make sure to edit the variable and change it back to denied.  And you will need to restart again.  Every time you change a variable, you need to restart.

**Comment by osheagrainne** [comment]
Great. Thank you. I have also put a budget in place and to be notified if it reaches that. I will test that and see how it works. 
Thank you again.

**Comment by Navid200** [comment]
What I suspect is that fitbit does not know how to use a token.  The best solution is to figure out a way to have fitbit use a token and change the setting back to denied.
But, let's go one step at a time.

**Comment by osheagrainne** [comment]
I was tempted to give building a clock face a go because other faces use the url fine but i love the look of this. I can always do the latter if the above doesn't work.

**Comment by Felguera242** [comment]
i am moving from heroku to google cloud too. Followed some tutorials to do it and i configured vm in cloud , install nightscout with xdrip script ,and now i can acces from browser and upload measures from diabox.  The problem is that when i use xdrip as follower ,and put the url in the NS follower box , a message appears denying acces "NS entries download was not succesful:401 unautorized" .I created a NS subjet and used this url   ,but doesnt work. But it works in nightwatch app (api key doesnt work in this app ,but subject token does. Its very weird , i am desesperated , i used xdrip to send data to my watch ,but now i cant.
I beg you to tell me wich tutorial did you follow to succeed . Thanks in advance.

**Comment by Felguera242** [comment]
i managed to "solve" it by changing AUTH_DEFAULT_RULES nightscout variable to "readable" in the virtual machine trough ssh .
Dont know if this is the right solution , but is a workaround solution for me until i know why xdrip doesnt access ns like it was doing with heroku.

**Comment by Navid200** [comment]
>  until i know why xdrip doesnt access ns like it was doing with heroku.

Heroku did not impose a limit on outgoing bandwidth.  On Google Cloud, if your outgoing bandwidth goes over 1GB/month, it will not be free.
We are not certain what will be the bandwidth and the resulting cost if crawlers also start reading if we open the site to the world for reading.

So, until we have more data, we decided to minimize the egress by default.

**Comment by Felguera242** [comment]
Thanks for your quick response. So I suppose this is a common problem for
all the people that is migrating from heroku to Google and not only mine.
We will wait for a solution that makes all people with less knowledge,
migrate without this problem. Thank you for your time and efforts.

**Comment by Navid200** [comment]
@Felguera242 Please don't quote an entire message unless you really want to.  If the reason your post looks like that is that you are responding to an email to post here, please don't.  It make the thread hard to follow.

The solution is to use apps and devices that know how to use tokens.  
But, if you must use a device that is incapable of using a token, you have to open to the world for read.  
I have no problem.

**Comment by Navid200** [comment]
@osheagrainne I just realized this was an issue.
I wish I had noticed earlier.
Please do not open an issue when you have a question, concern, or want to report a bug.  Instead please open a discussion.
